src/opencl_sha256crypt_fmt_plug.c

```diff
@@ -561,19 +572,25 @@ static int crypt_all(int *pcount, struct db_salt *_salt)
 		int tot_todo = 0, len;
 
 		if (lens_differ) {
-			indices = mem_alloc(gws * sizeof(int));
+
+			if (gws < previous_gws) {
```

Comment:
I think you got this check backwards - we need to reallocate if the previous size is no longer sufficient, not when it's excessive. Also drop the empty line above this check.

Reply:
You should review the code in this matter.